practice-two/src/components/FormInput/FormInput.jsx

```diff
@@ -0,0 +1,60 @@
+import { useState } from 'react';
+import './FormInput.css';
+import PropTypes from 'prop-types';
+import { MESSAGES, REGEX } from '@constants';
+
+const FormInput = (props) => {
+  const [focused, setFocused] = useState(false);
+  const { label, errorMessage, onChange, ...inputProps } = props;
+
+  const handleFocus = () => {
+    setFocused(true);
+  };
+
+  return (
+    <div className='form__input'>
+      <label className='input__label'>{label}</label>
+      <input
+        className='input__field'
+        {...inputProps}
+        onChange={onChange}
+        onBlur={handleFocus}
+        // eslint-disable-next-line react/no-unknown-property
+        focused={focused.toString()}
+      />
+      <span className='error__message'>{errorMessage}</span>
```

Comment:
if the error is empty the DOM will render `<span className='error__message'></span>`
Is it neccessary?

Reply:
I understood your comment
I changed that if having errorMessage and errorMessage is not empty string, it will render tag span to show errorMessage
![image](https://github.com/datdragon777/React-Training/assets/74806826/3e785f3e-7506-40b2-89a1-4155cc6a812a)